addons/binding/org.openhab.binding.kodi/src/main/java/org/openhab/binding/kodi/internal/protocol/KodiConnection.java

```diff
@@ -528,13 +530,19 @@ private void requestPlayerItemUpdate(int activePlayer) {
                 }
 
                 RawType thumbnail = null;
-                if (item.has("thumbnail")) {
-                    thumbnail = downloadImage(convertToImageUrl(item.get("thumbnail")));
+                if (item.has(PROPERTY_THUMBNAIL)) {
+                    String thumbnailURL = convertToImageUrl(item.get(PROPERTY_THUMBNAIL));
+                    if (thumbnailURL != null) {
```

Comment:
Would it make more sense to do the null check inside instead of repeating it?

Reply:
You could even consider having a method doing both the `convertToImageUrl` and `downloadImageFromCache`